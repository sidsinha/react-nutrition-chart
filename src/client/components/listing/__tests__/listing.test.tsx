import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Listing from './../listing';

const items = [
    {
        "id": 1,
        "name": "Oreo",
        "protein": 4,
        "fat": 18,
        "calories": 437,
        "carbs": 63
    },
    {
        "id": 2,
        "name": "Nougat",
        "protein": 37,
        "fat": 19,
        "calories": 308,
        "carbs": 9
    },
    {
        "id": 3,
        "name": "Marshmallow",
        "protein": 2,
        "fat": 0,
        "calories": 318,
        "carbs": 81
    }
]

afterEach(cleanup);

test('should render all todos', async() => {
    const { getByText } = render(<Listing itemList={items} status="sucess" />);
    for (const item of items) {
      expect(getByText(item.name)).toBeInTheDocument();
    }
  });

test('matches snapshot', async() => {
    const { container } = render(<Listing itemList={items} status="sucess" />)
    expect(container.firstChild).toMatchSnapshot();
  });