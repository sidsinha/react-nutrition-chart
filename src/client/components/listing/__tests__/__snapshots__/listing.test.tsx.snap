// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="items"
  id="item-listing"
>
  <div
    class="bg-white flex pv3 justify-around  bb b--moon-gray b"
  >
    <div
      class="ml3"
      style="flex: 0.25;"
    >
      <input
        type="checkbox"
      />
    </div>
    <div
      class=""
      style="flex: 1;"
    >
      Dessert(100g serving)
      <span
        class="pointer ml1 f6 blue"
      >
        ⇅
      </span>
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      Calories
      <span
        class="pointer ml1 f6 false"
      >
        ⇅
      </span>
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      Fat (g)
      <span
        class="pointer ml1 f6 false"
      >
        ⇅
      </span>
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      Carbs (g)
      <span
        class="pointer ml1 f6 false"
      >
        ⇅
      </span>
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      Protein (g)
      <span
        class="pointer ml1 f6 false"
      >
        ⇅
      </span>
    </div>
  </div>
  <div
    class="flex pv3 justify-around bb b--moon-gray false item"
  >
    <div
      class="ml3"
      style="flex: 0.25;"
    >
      <input
        data-testid="checkbox_nutrition"
        id="item-Oreo-checkbox"
        name="nutrition"
        type="checkbox"
        value="1"
      />
    </div>
    <div
      class="item-name"
      style="flex: 1;"
    >
      Oreo
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      437
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      18
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      63
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      4
    </div>
  </div>
  <div
    class="flex pv3 justify-around bb b--moon-gray false item"
  >
    <div
      class="ml3"
      style="flex: 0.25;"
    >
      <input
        data-testid="checkbox_nutrition"
        id="item-Nougat-checkbox"
        name="nutrition"
        type="checkbox"
        value="2"
      />
    </div>
    <div
      class="item-name"
      style="flex: 1;"
    >
      Nougat
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      308
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      19
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      9
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      37
    </div>
  </div>
  <div
    class="flex pv3 justify-around bb b--moon-gray false item"
  >
    <div
      class="ml3"
      style="flex: 0.25;"
    >
      <input
        data-testid="checkbox_nutrition"
        id="item-Marshmallow-checkbox"
        name="nutrition"
        type="checkbox"
        value="3"
      />
    </div>
    <div
      class="item-name"
      style="flex: 1;"
    >
      Marshmallow
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      318
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      0
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      81
    </div>
    <div
      class="tc"
      style="flex: 0.5;"
    >
      2
    </div>
  </div>
</div>
`;
